### PR TITLE
[IMP] point_of_sale,pos_restaurant: change table name system

### PR DESCRIPTION
**Pull request by davidmonnom**
Before, the table system was very basic and not very convenient for the user. When he wanted to create new table between floor plan, sometime several tables where created with the same number. This was due to the fact that the system was not able to detect the last table number between all floor plan.

Now each floor plan has a prefix and the system will name new table with the next number of the last table of the same floor plan.

taskId: 4283519

### CODE REVIEW

File: addons/point_of_sale/static/src/app/components/popups/number_popup/number_popup.xml [hunk 1]
```diff
@@ -10,10 +10,12 @@
                 </div>
             </div>
             <div t-if="props.feedback(this.state.buffer)" class="p-2 mx-auto mt-n2 mb-2" t-esc="props.feedback(this.state.buffer)" />
+            <t t-set="isValid" t-value="props.isValid(state.buffer)"/>
+            <div t-if="!isValid and props.isValidFeedback" class="p-2 mx-auto mt-n2 mb-2 text-center" t-esc="props.isValidFeedback(this.state.buffer)" />
             <Numpad buttons="props.buttons" class="'p-1 rounded-3 bg-200'"/>
             <t t-set-slot="footer">
                 <div class="d-flex flex-grow-1 justify-content-center">
-                    <button t-att-disabled="!props.isValid(state.buffer)" class="btn btn-primary btn-lg lh-lg o-default-button me-2" tabindex="1" t-on-click="confirm">
+                    <button t-att-disabled="!props.isValid(state.buffer) and props.isValidBlocking" class="btn btn-primary btn-lg lh-lg o-default-button me-2" tabindex="1" t-on-click="confirm">
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/components/popups/number_popup/number_popup.xml:18]: use the `isValid` that you set above

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

**Comment by adgu-odoo** [comment]
Also when I add tables into the bar (demo data) it always gives me tables named 101
image

File: addons/point_of_sale/static/src/app/components/popups/number_popup/number_popup.xml (hunk 1 above)
Comment (davidmonnom) [reply addons/point_of_sale/static/src/app/components/popups/number_popup/number_popup.xml:18]: I cannot use it in t-set the state isn't updated... :(

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
robodoo r+

**Comment by davidmonnom** [comment]
@adgu-odoo le seum

**Comment by adgu-odoo** [comment]
robodoo r+

**Pull request closed without merging**